### PR TITLE
Add hugo version check and theme update to Makefile

### PR DESCRIPTION
**Pull request by nginx-jack**
### Proposed changes

This should help prevent any hugo or hugo-theme version drift going forward, and warn when versions are out of date.

- Add hugo version check and theme update to Makefile
- Update README.md with correct make target behaviors
- Add hugo-entrypoint to allow extra commands when docker starts
- Update hugo theme version in go.mod
- Use slimmer hugo docker image

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -1,61 +1,56 @@
...
-    HUGO=docker run --rm -it -v ${CURDIR}:/src -p 1313:1313 ${HUGO_IMG} hugo --bind 0.0.0.0 -p 1313
-endif
+    $(warning Hugo is not available or using a version less than 134. Attempting to use docker. HUGO_VERSION=$(HUGO_VERSION))
+    HUGO=docker run --rm -it -v ${CURDIR}:/src -p 1313:1313 ${HUGO_IMG} /src/hugo-entrypoint.sh
+    ifeq (, $(shell docker version 2> /dev/null))
+        $(error Hugo (>0.134) or Docker are required to build the local previews.)
+    endif
 endif
 
-HUGO_CI=docker run --rm -v ${CURDIR}:/src ${HUGO_IMG} hugo
```
Comment (j1m-ryan) [inline docs/Makefile:15]: we use this in the CI, `make docs` which uses the `HUGO` variable doesn't work in the CI, because of it `-it` I think

Comment (j1m-ryan) [reply docs/Makefile:15]: https://github.com/nginxinc/kubernetes-ingress/blob/76651cc80eec84fa7cc7ffd29c2733996e1ee663/.github/workflows/ci.yml#L207

**Review by ADubhlaoich — APPROVED** [review]

File: docs/go.mod [hunk 2]
```diff
@@ -2,4 +2,4 @@ module github.com/nginxinc/kubernetes-ingress/docs
 
 go 1.19
```
Comment (vepatel) [inline docs/go.mod:3]: @nginx-jack can we update this to a more recent version?

File: docs/Makefile (hunk 1 above)
Comment (nginx-jack) [reply docs/Makefile:15]: Ah I didn't spot that. Thanks, will update 👍

File: docs/go.sum [hunk 3]
```diff
@@ -1,2 +1,4 @@
 github.com/nginxinc/nginx-hugo-theme v0.41.17 h1:KtF0TFQYXmVratjKVGEodWYFNbtxeOdrZLa9zi70BtA=
```
Comment (j1m-ryan) [inline docs/go.sum:1]: can you do a `make tidy`?

**Review by j1m-ryan — APPROVED** [review]

**Pull request merged**